### DISCUSSION TITLE
How to indicate external dependency in built package (metadata Requires-External, PEP725)

### DISCUSSION THREAD

**Discussion by BlueMagma2** [python-poetry/poetry#8534] [Q&A] [answered]
Since version 1.2 core-metadata support a Requires-External field to indicate to whatever installer that there are non-python dependencies and that the package might not work properly without those: https://packaging.python.org/en/latest/specifications/core-metadata/#requires-external-multiple-use

Draft PEP 725 proposes a pyproject.toml syntax that could be used to generate those line in the METADATA file: https://peps.python.org/pep-0725/

Currently is there any way to have poetry add a Requires-External line in my built package ? 
Or maybe a plugin exist for this ?

**Comment by radoering** [comment, marked as answer]
I don't think there is a built-in way. Metadata is handled in [poetry.core.masonry.metadata](https://github.com/python-poetry/poetry-core/blob/main/src/poetry/core/masonry/metadata.py) and as it looks `requires_external` is not set.

**Comment by BlueMagma2** [comment]
Thank you for your help, I ended up building a poetry plugin that reads the pyproject.toml and add the Requires-External during build.

I hope poetry will support it natively in the future

**Reply by povilasb** [reply]
@BlueMagma2 I'd be highly interested in that plugin - have you considered open sourcing it? :)

**Reply by BlueMagma2** [reply]
My employer agreed to make it public, so here it is: https://gitlab.com/public-corner/poetry-external-dependencies

Please note that in order to pip install the plugin, you need to use the --extra-index-url parameter to add the registry for this project since it is not published to pypi yet.
